linera-core/src/client.rs

```diff
@@ -1714,6 +1691,16 @@ where
         Ok(certificates)
     }
 
+    /// Creates an empty block to process all incoming messages. This may require several blocks.
+    /// If we are not a chain owner, this doesn't fail, and just returns an empty list.
+    pub async fn process_inbox_if_owned(&mut self) -> Result<Vec<Certificate>, ChainClientError> {
```

Comment:
Thanks! We could also try to predict instead if we have the key (but perhaps that's not easy from there).

Reply:
We could add another field to `struct ChainClient` to keep track of whether we are a leader or not.

But then I'd really like to group that together with `timestamp`, `block_hash` and `next_block_height` in a separate struct (so that all of them are a single field of `ChainClient`)… or reuse `ChainTipState`?
Anyway, I'd also prefer to do that in a separate PR.